CONTRIBUTING.md

```diff
@@ -150,6 +150,97 @@ cd test/manual/k8s_crds
 ```
 This test currently does not run as part of the test suite.
 
+## Profiling
+Profiling can be used to monitor the impact of Secretless on CPU and Memory consumption. Currently, Secretless supports two types- CPU and Memory.
+
+**Prerequisites:**
+- [Graphviz](https://graphviz.gitlab.io/download/) to visualize profiling results
+- [Postgresql](https://www.postgresql.org/download/) to install Postgres
+
+In order to configure individual profiles properly and receive desired results, perform the following:
+1. [Build](https://github.com/cyberark/secretless-broker/blob/master/CONTRIBUTING.md#building) Secretless locally
+2. Run a Postgres container named ```pg-test``` and wait for it to become ready:
+```
+docker run -d --name pg-test -p 5432:55432  postgres:9.3;
+
+# wait for pg-test
+docker exec pg-test bash -c "
+ while ! pg_isready -U postgres > /dev/null 2>&1;
+   do
+     >&2 echo "."
+     sleep 1
+ done
+ echo 'Postgres is up and ready'
+"
+```
+
+3. Create and populate a database named test:
+```
+docker exec pg-test psql -U postgres  -c "
+CREATE USER test PASSWORD 'test';
+
+CREATE SCHEMA test;
+CREATE TABLE test.test ( id INTEGER PRIMARY KEY );
+INSERT INTO test.test VALUES ( generate_series(0, 99999) );
+
+GRANT ALL ON SCHEMA test TO test;
+GRANT ALL ON test.test TO test;
+"
+```
+4. Create a sample secretless.yml file in the project root that has:
+
+```
+listeners:
+  - name: pg_tcp
+    protocol: pg
+    address: 0.0.0.0:15432
+
+handlers:
+  - name: pg_via_tcp
+    listener: pg_tcp
+    debug: true
+    credentials:
+      - name: address
+        provider: env
+        id: PG_ADDRESS
+      - name: username
+        provider: literal
+        id: test
+      - name: password
+        provider: env
+        id: PG_PASSWORD
+```
+5. The type of profiling is explicitly defined in the initial command that runs Secretless. Run Secretless with the profile desired like so:
+```
+$ PG_ADDRESS=localhost:5432/postgres PG_PASSWORD=test ./dist/darwin/amd64/secretless-broker -profile=cpu -f secretless.yml
```

Comment:
it might help to break this into a multiline command, like
```
PG_ADDRESS=localhost:5432/postgres \
  PG_PASSWORD=test \
  ./dist/darwin/amd64/secretless-broker \
  -profile=cpu \
  -f secretless.yml
```

Reply:
also, depending on your OS your binary may be located in a different folder

Reply:
how could we make this generic across the different OS? Can we make a "note:" at the bottom ?

Reply:
I think it's ok to leave it mac-speciifc - I'd just put a note that you need to use the binary for your OS. like `Note: we've used the Darwin binary in the examples below, but you should use the binary you built for your OS"